sdk/python/kfp/components/structures.py

```diff
@@ -574,6 +574,7 @@ def load_from_component_yaml(cls, component_yaml: str) -> 'ComponentSpec':
         Returns:
             Component spec in the form of V2 ComponentSpec.
         """
+
         json_component = yaml.safe_load(component_yaml)
         try:
             return ComponentSpec.parse_obj(json_component)
```

Comment:
Change L581 to `except Exception` would do the trick. We are trying if the item is v2, otherwise try v1.